### DISCUSSION TITLE
Add some alarm items

### DISCUSSION THREAD

**Discussion by litefrez** [frangoteam/FUXA#1393] [General]
I would like the ability to have a user adjust alarm setpoints without having access to the backend.  For instance, I have a 24 ft tall tank.  I want a high high at 22 ft today, but my user may want to adjust it to 21 ft or something along those lines temporarily during maintenance etc.  So if we could make a tag in the fuxa server to hold the highhigh level setpoint and then the alarm could compare against that tag.  
I would also like to be able to send alarms with telegram api or in the future some other http api.   I have already started the work and can send the alarm now, however my code may not be up to your coding standards.  I will need to spend some time looking at that.
I would be happy to help with the coding required etc., but I am less skilled with javascript.

**Comment by litefrez** [comment]
So, I believe I have the httpapi / telegram notification code finished, but there still may be some things I missed.  Basically with telegram api you can set up a group and add a bot to the group.  That will allow you to send messages by sending an https request like:  https://api.telegram.org/botxxxxxxxxxxxxxxxxxxxxxxxxxxxx/sendMessage?chat_id=-9xxxxxxxx&text=some text here
So after updating the notificator index.js file with the changes that I made, you would put everything from https://........text= into the receiver field and the edits would identify that it is not an email when it calls to send the notification and would send it as an https call instead.
I do not know how to use github and pull requests etc, but am happy to share what I have done.
I have attached the index.js file as index.txt
However I ran into a small issue that maybe some else has the answer to.  If there are more than one notification entities enabled, the second one seems to resend the alarm every minute or so.  I am not sure if this is a known issue, or something my code edits caused.
Any help or guidance would be appreciated.
Thanks
[index.txt](https://github.com/user-attachments/files/17048148/index.txt)

**Reply by unocelli** [reply]
Hi, Thanks a lot, let me try and if it works we will integrate it

**Reply by unocelli** [reply]
I'm working on this feature

**Comment by henjoe** [comment]
This is actually good features/refactoring. Most of scada thr alarms is integrated on a the Tag level. Fuxa is different, the alarm is created and you bind the Tag on that alarm. 

My suggestion is to refactor Tag model to include the AlarmSettings on the Tag itself. 


export Class AlarmSettings{
 enabled: Boolean,
 HH: alarmSetting,
 H: alarmSetting
 L: alarmSetting
}

Then add the UI on the Tag Options (separate Tab) for settin the Alarms. 

But these requires refactoring but achievable.
